### PR TITLE
Auction / Bid Endpoints

### PR DESCRIPTION
**Pull request by askabderon**
Auction and Bid Endpoints for Art Item is implemented in this PR.

Auction Endpoint:

- art_item/auction/{ArtItemId}

No additional body required for this endpoint.

This sets the art item with the given id on sale if it had not been setted already. If the given art item is already on sale, then removes from the sale. This is indicated with the onAuction field of Art Item. It is a boolean field; true means on sale and false means not on sale.

This requires the art item belonging to the user with the token.

- art_item/bid/{ArtItemId}

This requires a body that indicates the amount of bid:
```json
{
    "bidAmount": 30
}
```
First of all, we check whether the corresponding art item is on sale. If it is not, then throw error.

Then, it is checked whether the user had given a bid before, if so then update the previous bid amount. If not then creates a new Bid entity with the fields of bidAmount (Double), biddedArtItem (Art Item), and bidder (Registered User). 

I implemented a new DTO class for Bid in order to not return Art Item and return the RegisteredUserDTO as bidder.

### CODE REVIEW

**Comment by askabderon** [comment]
By latest commit, I add a new error if an artist try to bid their own art item.

File: backend/src/main/kotlin/com/group7/artshare/entity/ArtItem.kt [hunk 1]
```diff
@@ -39,10 +40,9 @@ class ArtItem{
     @Column
     var onAuction: Boolean = false
 
-    @OneToOne(cascade = [CascadeType.ALL])
-    @JoinColumn(name = "auction", referencedColumnName = "id")
+    @OneToMany(orphanRemoval = true, cascade = [CascadeType.ALL])
     @JsonIdentityInfo(generator = ObjectIdGenerators.PropertyGenerator::class, property = "id")
-    var auction: Auction? = null
+    var bids: MutableList<Bid> = mutableListOf()
```
Comment (sabrimete) [inline backend/src/main/kotlin/com/group7/artshare/entity/ArtItem.kt:45]: I think storing only the highest bid is enough. Others are not required.

**Review by sabrimete — CHANGES_REQUESTED** [review]
Wrote a comment to request a small change. Service function should also be updated accordingly. Other than that it looks fine.

File: backend/src/main/kotlin/com/group7/artshare/entity/ArtItem.kt (hunk 1 above)
Comment (askabderon) [reply backend/src/main/kotlin/com/group7/artshare/entity/ArtItem.kt]: Updated the code according to this. Thanks for pointing out.

File: backend/src/main/kotlin/com/group7/artshare/DTO/ArtItemDTO.kt [hunk 2]
```diff
@@ -19,7 +18,7 @@ class ArtItemDTO : Comparable<ArtItemDTO> {
     var ownerAccountInfo : AccountInfo? = null
     var ownerId : Long? = null
     var onAuction : Boolean? = null
-    var auction : Auction? = null
+    var maxBid : Bid? = null
```
Comment (sabrimete) [inline backend/src/main/kotlin/com/group7/artshare/DTO/ArtItemDTO.kt:21]: This should be BidDTO

**Review by sabrimete — APPROVED** [review]
LGTM, works on my local. Thanks for the work.

File: backend/src/main/kotlin/com/group7/artshare/DTO/ArtItemDTO.kt (hunk 2 above)
Comment (askabderon) [reply backend/src/main/kotlin/com/group7/artshare/DTO/ArtItemDTO.kt]: updated with the latest commit, thanks for pointing out.

**Pull request merged**